cmake/helpers.cmake

```diff
@@ -113,12 +116,23 @@ function(add_umf_target_compile_options name)
     elseif(MSVC)
         target_compile_options(
             ${name}
-            PRIVATE $<$<CXX_COMPILER_ID:MSVC>:/MP> # clang-cl.exe does not
-                                                   # support /MP
-                    /W3 /MD$<$<CONFIG:Debug>:d> /GS)
-
-        if(UMF_DEVELOPER_MODE)
-            target_compile_options(${name} PRIVATE /WX /GS)
+            PRIVATE # clang-cl.exe does not support /MP
+                    $<$<CXX_COMPILER_ID:MSVC>:/MP>
+                    /W4
+                    /WX
+                    /MD$<$<CONFIG:Debug>:d>
+                    /Gy
+                    /GS
+                    /analyze
+                    # disable warning 6326: Potential comparison of a constant
+                    # with another constant
+                    /wd6326
+                    # disable 4200 warning: nonstandard extension used:
+                    # zero-sized array in struct/union
+                    /wd4200
+                    /DYNAMICBASE)
+        if(CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
You can't rely on this variable during Windows builds. Probably `target_compile_options` supports this construct:
```
$<$<CONFIG:Release>:/sdl>
$<$<CONFIG:Release>:/LTCG>
$<$<CONFIG:Release>:/NXCOMPAT>
```

Reply:
done